Automattic/sensei
# Setup e2e tests using playwright

It resolves part of the #5107

### Changes proposed in this Pull Request
The main goal of this PR is to add a new e2e setup using the `playwright/test` package.
It is using the `Add to Course`  and `Remove From Course` scenarios because they don't require creating lessons inside the courses, simplifying the data setup. Lesson creation/completion requires further investigation about how to create the data properly using the API or PHP scripts.

**[Readme file](https://github.com/Automattic/sensei/blob/220a78232beff89372d53565f918d127fee8ba46/tests/e2e-playwright/README.md)**


#### Items added to this PR
* e2e setup local and CI
* Page Object Model (explained on the readme file)
* Run Authentication before the test suite (explained on the readme file)
* Create Courses and Students dynamically using the PR  (explained on the readme file)
* Store and Load Browser contexts by user role, now only admin is available but is easy to scale.  Explained on the readme file.
* Add a `legacy` prefix on the already existent e2e testing. 

### Testing Instructions
Please check the [Readme file](https://github.com/Automattic/sensei/pull/5102/commits/220a78232beff89372d53565f918d127fee8ba46#diff-ee922ceef65dc8e14eca60cca9512663565e411da4c6f64205e2ba077ad17988) to get instructions about how to run locally, it is a way to check if the instructions are complete.



#### Extra Notes
This PR is not using the [@wordpress/e2e-test-utils-playwright ](https://github.com/WordPress/gutenberg/tree/trunk/packages/e2e-test-utils-playwright ) because the package is still experimental and some of the implemented features were absorbed by @playwright/test, now there is no special feature we need from that package, keeping our setup as simple as possible.
